.github/workflows/generate-maven-sbom.yml

```
@@ -0,0 +1,71 @@
+name: Generate Maven SBOM
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    # Provide custom 'Version' input, to allow running the workflow for older
+    # git refs, where the workflow file did not exist yet. This is not possible
+    # with the builtin "Use workflow from" input field.
+    inputs:
+      version:
+        description: "Version"
+        default: "main"
+        required: true
+
+env:
+  JAVA_VERSION: '17'
+  JAVA_DISTRO: 'temurin'
+  PLUGIN_VERSION: '2.9.1'
+  SBOM_TYPE: 'makeAggregateBom'
+  PROJECT_VERSION: "${{ github.event_name == 'release' && github.event.release.tag_name || github.event.inputs.version }}"
+
+permissions:
+  contents: read
+
+jobs:
+  generate-sbom:
+    runs-on: ubuntu-latest
+    outputs:
+      # Make env var available in re-usuable workflow (see actions/runner#2372)
+      project-version: ${{ env.PROJECT_VERSION }}
+    steps:
+      - name: Checkout repository at '${{ env.PROJECT_VERSION }}'
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          fetch-depth: 0
+          ref: ${{ env.PROJECT_VERSION }}
+          persist-credentials: false
+
+      - name: Setup Java SDK
+        uses: actions/setup-java@dded0888837ed1f317902acf8a20df0ad188d165 # v5.0.0
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DISTRO }}
+
+      - name: Generate
+        run: |
```

You will want to patch the version, as you do [in your release workflow](https://github.com/eclipse-serializer/serializer/blob/e7f5425438e2ff38ba84f8db8595dc98501491b4/.github/workflows/maven_release.yml#L26), to list the correct project version in the sbom, e.g. `"3.0.1"` for the last release: 

```suggestion
        run: |
          if git show-ref --tags --quiet --verify "refs/tags/${PROJECT_VERSION}"; then
            mvn versions:set -DnewVersion=${PROJECT_VERSION} --batch-mode
          fi
```

The if-clause makes it so that if you trigger this workflow on a non-tag git reference such as a branch or commit id, then the sbom will list the version from pom.xml. Typically this is `"<NEXT MAJOR VERSION>-SNAPSHOT"`, currently `"4.0.0-SNAPSHOT"`.

You could probably remove the if-clause, then the git reference would be listed as version, e.g. `"main"` or "`zj/test-sbom-generation`". But I'm not sure, if the maven version plugin will allow any name.